core/src/main/java/hudson/util/io/TarArchiver.java

```diff
@@ -59,7 +59,7 @@
     public void visitSymlink(File link, String target, String relativePath) throws IOException {
         TarArchiveEntry e = new TarArchiveEntry(relativePath, TarConstants.LF_SYMLINK);
         try {
-            int mode = IOUtils.mode(link);
```

Comment:
This fix also should also render the "fix" that was made for JENKINS-13202 obsolete (actually, when archiving with a symlink-capable FileVisitor, links should **never** be resolved but archived as is. Also: When supporting symlinks, **never** set the permissions of a symlink to the permissions of its target.